File: tests/src/main/java/com/github/klee0kai/test/lifecycle/stone_util/LifecycleUtils.java
package com.github.klee0kai.test.lifecycle.stone_util;

import com.github.klee0kai.stone.types.lifecycle.IStoneLifeCycleOwner;
import com.github.klee0kai.test.lifecycle.base.APhone;
import com.github.klee0kai.test.lifecycle.base.APhoneLifecycle;

public class LifecycleUtils {

    public static IStoneLifeCycleOwner createFromPhone(APhone phone){
        return listener -> phone.subscribe(
                new APhoneLifecycle() {
                    @Override
                    public void onBuy() {

                    }

                    @Override
                    public void onBroken() {

                    }

                    @Override
                    public void onDrown() {
                        listener.protectForInjected(100);
                    }
                }
        );
    }


}

File: test_kotlin/src/main/kotlin/com/github/klee0kai/test/lifecycle/stone_util/LifecycleUtils.kt
package com.github.klee0kai.test.lifecycle.stone_util

import com.github.klee0kai.stone.types.lifecycle.IStoneLifeCycleListener
import com.github.klee0kai.stone.types.lifecycle.IStoneLifeCycleOwner
import com.github.klee0kai.test.lifecycle.base.APhone
import com.github.klee0kai.test.lifecycle.base.APhoneLifecycle

object LifecycleUtils {
    fun createFromPhone(phone: APhone): IStoneLifeCycleOwner {
        return IStoneLifeCycleOwner { listener: IStoneLifeCycleListener ->
            phone.subscribe(
                object : APhoneLifecycle {
                    override fun onBuy() {}
                    override fun onBroken() {}
                    override fun onDrown() {
                        listener.protectForInjected(100)
                    }
                }
            )
        }
    }
}